packages/core/src/contrib/Constraints.ts

```diff
@@ -36,23 +36,46 @@ import {
 import * as BBox from "engine/BBox";
 import { shapedefs } from "shapes/Shapes";
 import * as ad from "types/ad";
+import { StyleFunction } from "types/functions";
 
 // -------- Simple constraints
 // Do not require shape queries, operate directly with `ad.Num` parameters.
+const equal: StyleFunction = {
+  name: "equal",
+  documentation: "Require that the value `x` is equal to the value `y`",
+  arguments: [
+    { name: "x", description: "First value", type: "real" },
+    { name: "y", description: "Second value", type: "real" },
+  ],
+  function: (x: ad.Num, y: ad.Num) => {
```

Comment:
a bit sus to have the field be called `function` since GitHub syntax highlighting thinks it's a keyword